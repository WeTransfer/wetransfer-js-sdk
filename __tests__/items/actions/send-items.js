const routes = require('../../../src/config/routes');
const sendItemsAction = require('../../../src/items/actions/send-items');

describe('Send items action', () => {
  let sendItems = null;
  const mocks = {};
  beforeEach(() => {
    mocks.request = { send: jest.fn() };
    mocks.request.send.mockReturnValue(() =>
      Promise.reject(new Error('Network error.'))
    );

    sendItems = sendItemsAction({
      routes,
      request: mocks.request
    });
  });

  it('should throw an error if items are not provided', async () => {
    try {
      await sendItems();
    } catch (error) {
      expect(error).toMatchSnapshot();
    }
  });
});
