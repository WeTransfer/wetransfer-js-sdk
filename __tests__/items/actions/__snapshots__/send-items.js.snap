// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Send items action should throw an error if items are not provided 1`] = `[Error: There was an error when adding items to the transfer.]`;
